### PR TITLE
ppp: update to 2.5.1

### PR DESCRIPTION
**Pull request by DragonBluep**
The original PR: https://github.com/openwrt/openwrt/pull/12848

Notice:
* The default `ppp` package size (cortex-a53 variant) increased by ~10 KB, 137 KB --> 147 KB.
* The MPPE feature was supported by the upstream. However, the MPPC feature is missing. Upstream said [MPPC](https://github.com/ppp-project/ppp/blob/d5aeec65752d4a9b3bb46771d0b221c4a4a6539e/README.MPPE#L36-L38) is a patented algorithm, so maybe it's not a good idea to keep it in openwrt repository?
* I don't expect it to fix something, but this PR may have some help for these [issues](https://github.com/openwrt/openwrt/issues?q=is%3Aissue+is%3Aopen+%22Timeout+waiting+for+PADO+packets%22), who knows.

### CODE REVIEW

**Review by nasbdh9 — APPROVED** [review]
tested on x86_64 and flogic

**Comment by nbd168** [comment]
Dropping MPPC seems like a good idea to me too

**Comment by systemcrash** [comment]
I think the patent on MPPC has expired. But I guess that doesn't matter here if they dropped it upstream.

File: package/network/services/ppp/patches/400-simplify_kernel_checks.patch [hunk 1]
```diff
@@ -29,7 +29,7 @@ Signed-off-by: Jo-Philipp Wich <jo@mein.io>
```
Comment (hauke) [inline package/network/services/ppp/patches/400-simplify_kernel_checks.patch:29]: We should update the kernel version from `2,6,37` to something more recent like 5.15 or 6.6. There are some checks for kernel `if (kernel_version >= KVERSION(4,7,0))`.

I think we can drop this complete patch. How much bytes does it save?

Comment (hauke) [reply package/network/services/ppp/patches/400-simplify_kernel_checks.patch:29]: It only suppresses some error messages which indicate real errors on recent kernel versions.

**Comment by hauke** [comment]
Thank you for the PR, I hope that this will not introduce too many new problems.

**Comment by Neustradamus** [comment]
Thanks to @DragonBluep for this PPP 2.5.1 PR (and @icegood for PPP 2.5.0 origin) and @hauke for merging and all others participants @nasbdh9 (test), @nbd168, @systemcrash (@rsalvaterra, @mrkiko, @dchard, @BKPepe, @wongsyrone, @zdz2019, @Ansuel, @ptpt52, @curtdept, @enaess)!

Linked to:
- https://github.com/openwrt/openwrt/pull/12848
- https://github.com/ppp-project/ppp/issues/460
- https://github.com/ppp-project/ppp/issues/350
- https://github.com/ppp-project/ppp/issues/245
- https://github.com/ppp-project/ppp/issues/111

**Comment by ptpt52** [comment]
pptpd build failed agaist this new ppp version

**Comment by DragonBluep** [comment]
> pptpd build failed agaist this new ppp version

It seems that OpenWrt buildbot did not catch any build issues:
https://buildbot.openwrt.org/images/#/grid

**Comment by ptpt52** [comment]
pptpd build log:
```
aarch64-openwrt-linux-musl-gcc -DHAVE_CONFIG_H -I. -I.  -I/mnt/Data/Sources/openwrt/x-wrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/usr/include -I/mnt/Data/Sources/openwrt/x-wrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/include -I/mnt/Data/Sources/openwrt/x-wrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/include/fortify   -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/mnt/Data/Sources/openwrt/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0=pptpd-1.4.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -fno-builtin -Wall -DSBINDIR='"/usr/sbin"' -c bcrelay.c
bcrelay.c: In function 'mainloop.isra':
bcrelay.c:592:26: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  592 |           if ( (ipp_p->ip.protocol == IPPROTO_UDP) &&
      |                 ~~~~~~~~~^~~~~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:593:19: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  593 |                (((ntohl(ipp_p->ip.daddr)) & 0x000000ff) == 0x000000ff) &&
      |                  ~^~~~~~~~~~~~~~~~~~~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:594:26: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  594 |                (ipp_p->ip.ttl != 1) &&
      |                 ~~~~~~~~~^~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:595:83: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  595 |                (!((*IP_UDPPDU_CHECKSUM_MSB_PTR((unsigned char *)ipp_p+(4*ipp_p->ip.ihl)) == 0) &&
      |                                                                          ~~~~~~~~~^~~~
bcrelay.c:126:63: note: in definition of macro 'IP_UDPPDU_CHECKSUM_MSB_PTR'
  126 | #define IP_UDPPDU_CHECKSUM_MSB_PTR(udppdu) ((unsigned char *)(udppdu) + 6 )
      |                                                               ^~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:605:97: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  605 |               logstr_cntr = sprintf(logstr, "UDP_BroadCast(sp=%d,dp=%d) from: %s relayed to: ", ntohs(ipp_p->udp.source),
      |                                                                                                 ^~~~~~~~~~~~~~~~~~~~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:606:37: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  606 |                                     ntohs(ipp_p->udp.dest), iflistLogRToString(&(iflist[i]), i, &(cur_ifsnr[i])));
      |                                     ^~~~~~~~~~~~~~~~~~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:639:33: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  639 |                   ipp_p->ip.ttl = 1;
      |                   ~~~~~~~~~~~~~~^~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:643:35: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  643 |                   ipp_p->ip.daddr = iflist[j].bcast;
      |                   ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:649:24: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  649 |                       (ntohl(ipp_p->ip.saddr) != iflist[i].ifdstaddr))
      |                        ^~~~~~~~~~~~~~~~~~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:651:37: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  651 |                     ipp_p->ip.saddr = htonl(iflist[i].ifdstaddr);
      |                     ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
bcrelay.c:658:67: warning: array subscript 'struct packet[0]' is partly outside array bounds of 'unsigned char[1518]' [-Warray-bounds=]
  658 |                   udppdu = (unsigned char *)ipp_p + (4 * ipp_p->ip.ihl);
      |                                                          ~~~~~~~~~^~~~
bcrelay.c:438:17: note: object 'buf' of size 1518
  438 |   unsigned char buf[1518];
      |                 ^~~
aarch64-openwrt-linux-musl-gcc  -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/mnt/Data/Sources/openwrt/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0=pptpd-1.4.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -fno-builtin -Wall -DSBINDIR='"/usr/sbin"'  -L/mnt/Data/Sources/openwrt/x-wrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/usr/lib -L/mnt/Data/Sources/openwrt/x-wrt/staging_dir/toolchain-aarch64_cortex-a53_gcc-13.3.0_musl/lib -fuse-ld=bfd -znow -zrelro  -o bcrelay bcrelay.o  
for d in plugins; do make -w -C $d all; done
make[5]: Entering directory '/mnt/build/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0/plugins'
aarch64-openwrt-linux-musl-gcc -o pptpd-logwtmp.so -shared -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/mnt/Data/Sources/openwrt/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0=pptpd-1.4.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -I.. -I../../include -fPIC pptpd-logwtmp.c -lutil
pptpd-logwtmp.c:17:23: error: 'VERSION' undeclared here (not in a function)
   17 | char pppd_version[] = VERSION;
      |                       ^~~~~~~
pptpd-logwtmp.c:19:31: error: 'PATH_MAX' undeclared here (not in a function)
   19 | static char pptpd_original_ip[PATH_MAX+1];
      |                               ^~~~~~~~
pptpd-logwtmp.c:22:8: error: unknown type name 'option_t'
   22 | static option_t options[] = {
      |        ^~~~~~~~
pptpd-logwtmp.c:23:3: warning: braces around scalar initializer
   23 |   { "pptpd-original-ip", o_string, pptpd_original_ip,
      |   ^
pptpd-logwtmp.c:23:3: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:23:5: warning: initialization of 'int' from 'char *' makes integer from pointer without a cast [-Wint-conversion]
   23 |   { "pptpd-original-ip", o_string, pptpd_original_ip,
      |     ^~~~~~~~~~~~~~~~~~~
pptpd-logwtmp.c:23:5: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:23:5: error: initializer element is not computable at load time
pptpd-logwtmp.c:23:5: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:23:26: error: 'o_string' undeclared here (not in a function)
   23 |   { "pptpd-original-ip", o_string, pptpd_original_ip,
      |                          ^~~~~~~~
pptpd-logwtmp.c:23:26: warning: excess elements in scalar initializer
pptpd-logwtmp.c:23:26: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:23:36: warning: excess elements in scalar initializer
   23 |   { "pptpd-original-ip", o_string, pptpd_original_ip,
      |                                    ^~~~~~~~~~~~~~~~~
pptpd-logwtmp.c:23:36: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:24:5: warning: excess elements in scalar initializer
   24 |     "Original IP address of the PPTP connection",
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pptpd-logwtmp.c:24:5: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:25:5: error: 'OPT_STATIC' undeclared here (not in a function)
   25 |     OPT_STATIC, NULL, PATH_MAX },
      |     ^~~~~~~~~~
pptpd-logwtmp.c:25:5: warning: excess elements in scalar initializer
pptpd-logwtmp.c:25:5: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:25:17: warning: excess elements in scalar initializer
   25 |     OPT_STATIC, NULL, PATH_MAX },
      |                 ^~~~
pptpd-logwtmp.c:25:17: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:25:23: warning: excess elements in scalar initializer
   25 |     OPT_STATIC, NULL, PATH_MAX },
      |                       ^~~~~~~~
pptpd-logwtmp.c:25:23: note: (near initialization for 'options[0]')
pptpd-logwtmp.c:26:3: warning: braces around scalar initializer
   26 |   { "pptpd-logwtmp-strip-domain", o_bool, &pptpd_logwtmp_strip_domain,
      |   ^
pptpd-logwtmp.c:26:3: note: (near initialization for 'options[1]')
pptpd-logwtmp.c:26:5: warning: initialization of 'int' from 'char *' makes integer from pointer without a cast [-Wint-conversion]
   26 |   { "pptpd-logwtmp-strip-domain", o_bool, &pptpd_logwtmp_strip_domain,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
pptpd-logwtmp.c:26:5: note: (near initialization for 'options[1]')
pptpd-logwtmp.c:26:5: error: initializer element is not computable at load time
pptpd-logwtmp.c:26:5: note: (near initialization for 'options[1]')
pptpd-logwtmp.c:26:35: error: 'o_bool' undeclared here (not in a function); did you mean 'bool'?
   26 |   { "pptpd-logwtmp-strip-domain", o_bool, &pptpd_logwtmp_strip_domain,
      |                                   ^~~~~~
      |                                   bool
pptpd-logwtmp.c:26:35: warning: excess elements in scalar initializer
pptpd-logwtmp.c:26:35: note: (near initialization for 'options[1]')
pptpd-logwtmp.c:26:43: warning: excess elements in scalar initializer
   26 |   { "pptpd-logwtmp-strip-domain", o_bool, &pptpd_logwtmp_strip_domain,
      |                                           ^
pptpd-logwtmp.c:26:43: note: (near initialization for 'options[1]')
pptpd-logwtmp.c:27:5: warning: excess elements in scalar initializer
   27 |     "Strip domain from username before logging", OPT_PRIO | 1 },
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pptpd-logwtmp.c:27:5: note: (near initialization for 'options[1]')
pptpd-logwtmp.c:27:50: error: 'OPT_PRIO' undeclared here (not in a function)
   27 |     "Strip domain from username before logging", OPT_PRIO | 1 },
      |                                                  ^~~~~~~~
pptpd-logwtmp.c:27:50: warning: excess elements in scalar initializer
pptpd-logwtmp.c:27:50: note: (near initialization for 'options[1]')
pptpd-logwtmp.c:28:3: warning: braces around scalar initializer
   28 |   { NULL }
      |   ^
pptpd-logwtmp.c:28:3: note: (near initialization for 'options[2]')
pptpd-logwtmp.c:28:5: warning: initialization of 'int' from 'void *' makes integer from pointer without a cast [-Wint-conversion]
   28 |   { NULL }
      |     ^~~~
pptpd-logwtmp.c:28:5: note: (near initialization for 'options[2]')
pptpd-logwtmp.c: In function 'ip_up':
pptpd-logwtmp.c:45:23: error: 'peer_authname' undeclared (first use in this function); did you mean 'ppp_peer_authname'?
   45 |   char *user = reduce(peer_authname);
      |                       ^~~~~~~~~~~~~
      |                       ppp_peer_authname
pptpd-logwtmp.c:45:23: note: each undeclared identifier is reported only once for each function it appears in
pptpd-logwtmp.c:46:7: error: 'debug' undeclared (first use in this function); did you mean 'debug_on'?
   46 |   if (debug)
      |       ^~~~~
      |       debug_on
pptpd-logwtmp.c:47:47: error: 'ifname' undeclared (first use in this function)
   47 |     notice("pptpd-logwtmp.so ip-up %s %s %s", ifname, user,
      |                                               ^~~~~~
pptpd-logwtmp.c:49:3: warning: implicit declaration of function 'logwtmp' [-Wimplicit-function-declaration]
   49 |   logwtmp(ifname, user, pptpd_original_ip);
      |   ^~~~~~~
pptpd-logwtmp.c: In function 'ip_down':
pptpd-logwtmp.c:54:7: error: 'debug' undeclared (first use in this function); did you mean 'debug_on'?
   54 |   if (debug)
      |       ^~~~~
      |       debug_on
pptpd-logwtmp.c:55:43: error: 'ifname' undeclared (first use in this function)
   55 |     notice("pptpd-logwtmp.so ip-down %s", ifname);
      |                                           ^~~~~~
pptpd-logwtmp.c: In function 'plugin_init':
pptpd-logwtmp.c:61:3: warning: implicit declaration of function 'add_options' [-Wimplicit-function-declaration]
   61 |   add_options(options);
      |   ^~~~~~~~~~~
pptpd-logwtmp.c:62:3: warning: implicit declaration of function 'add_notifier' [-Wimplicit-function-declaration]
   62 |   add_notifier(&ip_up_notifier, ip_up, NULL);
      |   ^~~~~~~~~~~~
pptpd-logwtmp.c:62:17: error: 'ip_up_notifier' undeclared (first use in this function)
   62 |   add_notifier(&ip_up_notifier, ip_up, NULL);
      |                 ^~~~~~~~~~~~~~
pptpd-logwtmp.c:63:17: error: 'ip_down_notifier' undeclared (first use in this function)
   63 |   add_notifier(&ip_down_notifier, ip_down, NULL);
      |                 ^~~~~~~~~~~~~~~~
pptpd-logwtmp.c:64:7: error: 'debug' undeclared (first use in this function); did you mean 'debug_on'?
   64 |   if (debug)
      |       ^~~~~
      |       debug_on
Makefile:19: recipe for target 'pptpd-logwtmp.so' failed
make[5]: *** [pptpd-logwtmp.so] Error 1
make[5]: Leaving directory '/mnt/build/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0/plugins'
Makefile:840: recipe for target 'all-local' failed
make[4]: *** [all-local] Error 2
make[4]: Leaving directory '/mnt/build/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0'
Makefile:277: recipe for target 'all' failed
make[3]: *** [all] Error 2
make[3]: Leaving directory '/mnt/build/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0'
Makefile:69: recipe for target '/mnt/Data/Sources/openwrt/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0/.built' failed
make[2]: *** [/mnt/Data/Sources/openwrt/x-wrt/build_dir/target-aarch64_cortex-a53_musl/pptpd-1.4.0/.built] Error 2
make[2]: Leaving directory '/mnt/Data/Sources/openwrt/openwrt-feeds/packages/net/pptpd'
time: package/feeds/packages/pptpd/compile#2.44#0.92#4.12
    ERROR: package/feeds/packages/pptpd failed to build.
package/Makefile:175: recipe for target 'package/feeds/packages/pptpd/compile' failed
make[1]: *** [package/feeds/packages/pptpd/compile] Error 1
make[1]: Leaving directory '/mnt/Data/Sources/openwrt/x-wrt'
/mnt/Data/Sources/openwrt/x-wrt/include/toplevel.mk:246: recipe for target 'package/feeds/packages/pptpd/compile' failed
make: *** [package/feeds/packages/pptpd/compile] Error 2

```

File: package/network/services/ppp/patches/400-simplify_kernel_checks.patch (hunk 1 above)
Comment (DragonBluep) [reply package/network/services/ppp/patches/400-simplify_kernel_checks.patch:29]: This patch can reduce the size of the package `ppp` by approximately 2.5 KB. I'll rewrite it to remove more useless checks later.

**Comment by curtdept** [comment]
@ptpt52 likely have to make an issue in packages but it looks like pptpd has had a 1.5 update out for a while that might address a bunch of this

https://sourceforge.net/p/poptop/git/ci/1.5.0/tree/

**Comment by ptpt52** [comment]
seems good to upgrade to pptpd 1.5.0

**Comment by Neustradamus** [comment]
There is a pptpd patch published 2023-08-08 by @yarda (Fedora/RedHat) here:
- https://sourceforge.net/p/poptop/patches/21/

Maybe you can comment on the patch ticket and request to @pghmcfc, @quozl from poptop team:
- https://sourceforge.net/p/poptop/_members/

**Comment by quozl** [comment]
> There is a pptpd patch published 2023-08-08 by @yarda (Fedora/RedHat) here:
> 
>     * https://sourceforge.net/p/poptop/patches/21/
> 
> 
> Maybe you can comment on the patch ticket and request to @pghmcfc, @quozl from poptop team:
> 
>     * https://sourceforge.net/p/poptop/_members/

Noted, no action required.  Please do upgrade to pptpd-1.5.0, and if any further changes are required rebase against HEAD and make pull request.  https://github.com/quozl/pptpd

**Comment by Neustradamus** [comment]
@DragonBluep: There is a new PPP version: 2.5.2, can you look?
Thanks in advance.

**Pull request merged**